src/main/java/eu/donals/GameState.java

```diff
@@ -0,0 +1,80 @@
+package eu.donals;
+
+import org.json.simple.JSONArray;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class GameState {
+
+    private Player player;
+    private Location location;
+    private JSONArray locationList;
+    private boolean isFinished;
+    private int numRedo;
+
+    public GameState updateGameState(InputParser parsedInput) throws IOException {
+        GameState newGameState = this;
+
+        switch (parsedInput.getCommand()) {
```

Comment:
Great choice using a switch statement! This is a nice structure that clearly states all the possible options.